candle-core/src/quantized/k_quants.rs

```diff
@@ -821,10 +822,16 @@ pub fn matmul<T: GgmlType>(
     for row_idx in 0..m {
         let lhs_row = &lhs_b[row_idx * k_in_lhs_blocks..(row_idx + 1) * k_in_lhs_blocks];
         let dst_row = &mut dst[row_idx * n..(row_idx + 1) * n];
-        for (col_idx, dst) in dst_row.iter_mut().enumerate() {
-            let rhs_col = &rhs_t[col_idx * k_in_rhs_blocks..(col_idx + 1) * k_in_rhs_blocks];
-            *dst = T::vec_dot(k, rhs_col, lhs_row)?;
-        }
+
+        dst_row
+            .into_par_iter()
+            .enumerate()
+            .with_min_len(128)
+            .with_max_len(512)
+            .for_each(|(col_idx, dst)| {
+                let rhs_col = &rhs_t[col_idx * k_in_rhs_blocks..(col_idx + 1) * k_in_rhs_blocks];
+                *dst = T::vec_dot(k, rhs_col, lhs_row).unwrap();
```

Comment:
Unwrap doesn't look so good here, if there is an error, we would want to report it properly rather than crashing.

Reply:
Ok, if this has any potential performance impact, let's maybe forget about it. The plan would be to check for these conditions outside of the hot loop at some point anyway so having an unwrap in the meantime wouldn't be that bad.

Reply:
Performance basically hasen't changed after implementing it, but i guess the entire `qmatmul` has to be optimized at some point.  We can keep it in for now.